decidim-api/app/controllers/decidim/api/sessions_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Api
+    class SessionsController < ::Devise::SessionsController
+      skip_before_action :verify_authenticity_token
+
+      respond_to :json
+
+      def failure
+        render json: anonymous_user.serializable_hash.merge(
+          "jwt_token" => nil,
+          "avatar" => nil
+        ), status: :unauthorized
+      end
+
+      private
+
+      def auth_options
+        { scope: resource_name, recall: "#{controller_path}#failure" }
+      end
+
+      def after_sign_in_path_for(_resource_or_scope)
+        nil
+      end
+
+      def respond_with(resource, _opts = {})
+        if request.env[::Warden::JWTAuth::Middleware::TokenDispatcher::ENV_KEY]
+          jwt_token = request.env[::Warden::JWTAuth::Hooks::PREPARED_TOKEN_ENV_KEY]
+          status = jwt_token ? 200 : 403
```

Comment:
Although the user is already authenticated at this point, we should not return the details for the user in case the JWT token could not be issued. The reason for this failure would be that the JWT secret has not been configured correctly.
```suggestion
          return failure if jwt_token.blank?
```